apps/src/aichat/views/modelCustomization/PromptCustomization.tsx

```diff
@@ -1,49 +1,76 @@
-import React, {useState, ChangeEvent} from 'react';
+import React from 'react';
+import {useSelector} from 'react-redux';
 
 import {StrongText} from '@cdo/apps/componentLibrary/typography/TypographyElements';
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+import {AichatLevelProperties} from '@cdo/apps/aichat/types';
 import styles from '../model-customization-workspace.module.scss';
+import {isVisible, isDisabled} from './utils';
+import {EMPTY_AI_CUSTOMIZATIONS} from './constants';
 
 const PromptCustomization: React.FunctionComponent = () => {
-  const [temperature, setTemperature] = useState('0.05');
+  const {botName, temperature, systemPrompt} = useSelector(
+    (state: {lab: LabState}) =>
+      (state.lab.levelProperties as AichatLevelProperties | undefined)
+        ?.initialAiCustomizations || EMPTY_AI_CUSTOMIZATIONS
+  );
 
-  const handleTemperatureChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setTemperature(event.target.value);
-  };
+  const allFieldsDisabled =
+    isDisabled(botName.visibility) &&
+    isDisabled(temperature.visibility) &&
+    isDisabled(systemPrompt.visibility);
 
   return (
     <div className={styles.verticalFlexContainer}>
       <div>
-        <div className={styles.inputContainer}>
-          <label htmlFor="chatbot-name">
-            <StrongText>Chatbot name</StrongText>
-          </label>
-          <input id="chatbot-name" />
-        </div>
-        <div className={styles.inputContainer}>
-          <div className={styles.horizontalFlexContainer}>
-            <label htmlFor="temperature">
-              <StrongText>Temperature</StrongText>
+        {isVisible(botName.visibility) && (
+          <div className={styles.inputContainer}>
```

Comment:
One idea - would it be a little cleaner to handle the visibility in CSS with classnames? Something like  `<div className={classNames(styles.inputContainer, isHidden(botName.visibility) && moduleStyles.hidden)}>`, where `.hidden` is a class that just applies `display: none`? This way is fine too so no strong preference, but just had the thought while looking at this.

Reply:
mmm yeah no strong preference either, I think I'll leave as-is for now but can definitely update down the line if we prefer.